### ISSUE TITLE
Espresso RecyclerViewActions.scrollTo issues

### ISSUE THREAD

**Issue by madsbf** [evant/binding-collection-adapter#72]
Consider the following code in an Espresso test:
```java
RecyclerViewActions.scrollTo(hasDescendant(withText(someValue)));
```

The _scrollTo_-method will go through each item in the adapter one by one and bind to the ViewHolder until it finds a View that fulfills the requirement _hasDescendant(withText(someValue))_. When the Adapter is a _BindingRecyclerViewAdapter_, the binding is however never executed, so the _scrollTo_-method will never work.
I looked into this, and found out, that the reason for this, is the following method in the _OnRebindCallback_:

```java
public boolean onPreBind(ViewDataBinding binding) {
    return recyclerView != null && recyclerView.isComputingLayout();
}
```

Since we are now trying to bind Views while the RecyclerView is not computing it's layout, this method always returns false, and the binding execution is halted.

It shouldn't be too hard for me to make a workaround, that kicks in when running the Espresso tests, but I'm wondering if it would be possible to make a more general approach. 
Anyway, I will be looking into a solution for this, but wanted to let you and anyone else experiencing this issue know.

**Comment by rpattabi** [comment]
Anyone successful in making espresso RecyclerViewActions work?

**Comment by madsbf** [comment]
Yes, I made it work with. I will share a Gist when I get to my laptop.

**Comment by madsbf** [comment]
https://gist.github.com/madsbf/895228e6e5655c22c4913d49fdd5c545

I use reflection to set mLayoutOrScrollCounter to 1 on the RecyclerView before performing RecyclerViewActions.scrollTo(). Hope this can help you :)

**Comment by rpattabi** [comment]
@madsbf It works great! Thanks for sharing.

In my case, I needed couple of steps to make it work for me.

1. I am still on binding-collection-adapter v1.3.0, so I changed package import from `me.tatarka.bindingcollectionadapter2.BindingRecyclerViewAdapter` to `me.tatarka.bindingcollectionadapter.BindingRecyclerViewAdapter`

2. I had minifyEnabled on even for debug build (forced due to Guava). This stripped out RecyclerView's `scrollToPosition` which made `scrollTo` to fail. To prevent this from happening, I had to add the following lines to my debug specific proguard config:

```
-keep class android.support.v7.widget.** { *; }
-keepclassmembers class android.support.v7.widget.** { *; <fields>; }
```

proguard probably stripped the code since it didn't catch the use of this through reflection.

**Comment by siavashabdoli** [comment]
I'm using default Recyclerview with android binding and above gist does not work for me;
I wrote another gist. I think it works in more cases
https://gist.github.com/siavashabdoli/cc503ca824c4fa922f453a711b14895f

**Comment by evant** [comment]
Has anyone reported an issue against the espresso lib for this? Seems like it shouldn't be calling the adapter in a different way than is expected when using in a recyclerview.
